### DISCUSSION TITLE
Increasing open file handle limit on Linux

### DISCUSSION THREAD

**Discussion by DongYuLiu-Boss** [rabbitmq/rabbitmq-server#7756] [Questions]
The RabbitMQ version used is 3.8.3, the Erlang version is 22.3.4.1, and the deployment environment is Linux; The number of File descriptors is 1024, and the number of Socket descriptors is 832. Once these two numbers are too high, RabbitMQ will be unable to connect, and the management side page will not respond. It seems that you can add the LimitNOFILE parameter to the rabbitmq-server.service configuration file to specify the number of connections. However, at present, I cannot modify this file and display an error E212: Unable to open and write to the file; How can I solve it? Did you specify the running parameters when starting the RabbitMQ service through systemctl to solve this problem?

**Comment by michaelklishin** [comment]
@DongYuLiu-Boss it's not nice to mention individual core team members a few minutes after you open a discussion. We are not obligated to provide instant support for free, or at all.

Nearly all modern Linux distributions use systemd. The way you bump the open file handle limit for a service such as RabbitMQ is [via systemd](https://rabbitmq.com/install-rpm.html#kernel-resource-limits). There is no need to edit service files and `ulimit -n` very likely won't have any effect beyond your active shell.

RabbitMQ 3.8 has been entirely out of support for [about eight months now](https://rabbitmq.com/versions.html). Even 3.9 is now out of community support.

**Reply by DongYuLiu-Boss** [reply]
Thank you for your reply. I took the initiative to delete it after I realized my bad behavior@

**Reply by DongYuLiu-Boss** [reply]
So, sir, would you please advise me how to write the complete instructions for starting RabbitMQ through systemd to increase the limit on opening file handles for services such as RabbitMQ?

**Reply by michaelklishin** [reply]
[The docs have an example](https://rabbitmq.com/install-rpm.html#kernel-resource-limits): edit `/etc/systemd/system/rabbitmq-server.service.d/limits.conf` to include

``` ini
[Service]
LimitNOFILE=64000
```

64K is a good starting number but on Erlang 22 (and even 25 IIRC) it would pre-allocate 64 times more memory for the file descriptor table than with 1024. So a value around 16K or 32K may be more optimal depending on how many connections you expect to have at usage peaks.

**Reply by DongYuLiu-Boss** [reply]
Thank you, sir. I apologize again for my rudeness. I will try as you say
